src/App.jsx

```diff
@@ -29,17 +30,23 @@ function App() {
 
   const [isOpen, setOpen] = useState(false);
   const [isDeleteOpen, setIsDeleteOpen] = useState(false);
-  
-  const toggleDialog = () => setOpen(!isOpen);
-  const toggleDeleteDialog = () => setIsDeleteOpen(!isDeleteOpen);
+
+  const toggleDialog = () => {
+    if (isOpen) setMovieToEdit();
```

Comment:
Bad approach... to many thing to be aware of.
It's better to separate create & edit dialogs